src/api/API.js

```diff
@@ -47,6 +47,28 @@ const API = {
     return fetch(relURL('/users/login'), postOptions(payload))
       .then(handleResponse);
   },
+
+  /**
+   * Logs user into account.
+   *
+   * @param {String} firstname
+   * @param {String} lastname
+   * @param {String} username
+   * @param {String} password
+   * @throws {APIError} On server error.
+   * @returns {Promise<UserResponse>}
+   */
+  register: async (firstname, lastname, email, username, password) => {
```

Comment:
Since this function takes quite a bit of parameters, it may be better to have it take an object instead, like
```js
register({
  firstName: 'foo',
  lastName: 'bar',
  ...
})
```
This way it'll be easier to change the function params in the future if we need to.

Reply:
As a corollary, you can then pass in the `data.entries()` into the function.

Reply:
I went with passing it as an object but not exactly sure if I did it as intended.

Reply:
I think you should changes the names on the inputs to match the ones taken by the method.

I was think something like:
```js
const formData = new FormData(event.target);
const data = Object.fromEntries(formData);

...

register(data) ...
```

Reply:
Made the changes.